server/apps/legal_archive/service.py

```diff
@@ -8,10 +8,45 @@
 # AUTHORS and LICENSE files distributed with this source code, or
 # at https://www.sourcefabric.org/superdesk/license
 
-from superdesk.services import BaseService
+import logging
+from superdesk import Service, get_resource_privileges
+from flask import g
+from superdesk.errors import SuperdeskApiError
 
 
-class LegalArchiveVersionsService(BaseService):
+logger = logging.getLogger(__name__)
+
+
+class LegalService(Service):
+
+    def get(self, req, lookup):
+        self.check_get_access_privilege()
+        return super().get(req, lookup)
+
+    def check_get_access_privilege(self):
+        if not hasattr(g, 'user'):
+            return
+
+        privileges = g.user.get('privileges', {})
```

Comment:
`auth.py` sets `active_privileges` on `g.user` not `privileges`. So privileges variable will always be {}. Please refer https://github.com/superdesk/superdesk/blob/master/server/apps/users/services.py#L296